mbtools/html_content_extractor.py

```diff
@@ -0,0 +1,47 @@
+from mbtools import utils
+from pathlib import Path
+import argparse
+
+
+class ContentExtractor:
+    """
+    Given a string with path to an extracted moodle backup directory return
+    model objects for course html elements
+    """
+    def __init__(self, mbz_path):
+        self.mbz_path = Path(mbz_path)
+        self.output_html_files = {}
+
+    def replace_content_tags(self, output_file_path):
+        activities = utils.parse_backup_activities(self.mbz_path)
+
+        for act in activities:
+            for html_elem in act.html_elements():
+
+                self.output_html_files.update(html_elem.replace_content_tag())
+                with open(act.activity_filename, "wb") as f:
+                    act.etree.write(f, encoding="utf-8", xml_declaration=True)
+
+        self.write_html_files(output_file_path)
+        return self.output_html_files
+
+    def write_html_files(self, output_file_path):
+        for file_name, file_content in self.output_html_files.items():
+            with open(f"{output_file_path}/{file_name}.html", "w") as file:
+                file.write(file_content)
+
+
+def main():
```

Comment:
It would be nice to have test coverage of `main()`. You can look at a few examples of how to do this such as [here](https://github.com/openstax/raise-mbtools/blob/3975438c99f18dfd3e5053c80e320a77639b35c0/tests/test_resource_upload.py#L176-L180) in our other tests.

Reply:
I added the tests and added a feature to allow user to pass in optional filters to the CLI. It extracts both page and lesson content if no filters is passed.  Do we want the filter to be a required argument?

Reply:
:thinking: - Having a reasonable default seems okay to me.